eZ/Publish/Core/Persistence/Cache/InMemory/InMemoryCache.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+/**
+ * File containing MetadataCachePool class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Persistence\Cache\InMemory;
+
+/**
+ * Simple In-Memory Cache Pool.
+ *
+ * Could be used as separate instances for different use cases with a shared global expiry when applicable.
+ *
+ * Allow to use one instance with low ttl (200-600 milliseconds) and limited items (~100) dedicated to content (and
+ * related classes) to handle burst caching, and a separate instance with longer ttl (3-10sec) and separate limit (~200)
+ * for meta values like content type, section, ... to handle repeated lookups for these seldom changing values.
+ *
+ * @internal Only for use in eZ\Publish\Core\Persistence\Cache\AbstractInMemoryHandler, may change depending on needs there.
+ */
+class InMemoryCache
+{
+    /**
+     * @var float Cache Time to Live, in seconds. This is only for how long we keep cache object around in-memory.
+     */
+    private $ttl;
+
+    /**
+     * @var int The limit of objects in cache pool at a given time
+     */
+    private $limit;
```

Comment:
Without going overboard: it would be nice to be able to programmatically set the limit for each cache pool, as this could be tuned both per-server as well as per-cript (thinking about cli scripts here mostly. The same probably applies to TTL

Reply:
ref `internal` flag... There should rather be a service for disabling cache imho which abstract all caches. As you can see inline there is a global config for tuning it, and the ttl and limit is there to make sure CLI does not use stale data for too long.